log/log.go

```diff
@@ -102,8 +102,10 @@ func StartObserver(o Options) error {
 	} else {
 		if val, ok := os.LookupEnv("KUBEARMOR_SERVICE"); ok {
 			gRPC = val
-		} else {
+		} else if fileContent, err := os.ReadFile("/opt/kubearmor/kubearmor.pid"); err != nil || string(fileContent) == "" {
```

Comment:
Few things:
1. we should check for the presence of this file only in non-k8s env.
2. What happens if a stray pid file lingers after the kubearmor is uncleanly stopped?

Reply:
>     1. we should check for the presence of this file only in non-k8s env.
> 
>     2. What happens if a stray pid file lingers after the kubearmor is uncleanly stopped?
@nthnieljson Can you please take a look at it?

Reply:
1. this is the same behavior as before, if there is no file (in non-k8s env), it will chooses the default value, which is "localhost:32767"
2. Then it will use the previous port value, this would cause problem, I am not really sure how to address this, need your input and help.